src/transformers/pipelines/__init__.py

```diff
@@ -727,28 +733,48 @@ def pipeline(
     if isinstance(model, Path):
         model = str(model)
 
+    if hub_kwargs["_commit_hash"] is None:
```

Comment:
You have isolated it in a separate variable so you could just do 
```suggestion
    if commit_hash is None:
```

Reply:
That variable is mainly to be used to update

```bash
    hub_kwargs = {
        ....
        "_commit_hash": commit_hash,
    }
```
and anything after is best to continue to use `hub_kwargs["_commit_hash"]`.

I can avoid the definition of `commit_hash` and just use `hub_kwargs["_commit_hash"]` everywhere.

Reply:
done locally and pushed 😅